concepts/infrastructure-reconciliation/README.md

```diff
@@ -1,41 +1,183 @@
+
+### Subscription Manager and the SubscriptionRequest CR
+
+This is a part that needs to be extracted from the Kyma Environment Broker, there are no significant changes in this area. The main change is introducing a new declarative API in form of the SubscriptionRequest CR:
+```
+apiVersion: control-plane.kyma-project.io/v1alpha1
+kind: SubscriptionRequest
+metadata:
+    name: subscription-request
+    namespace: cluster-manager
+spec:
+    provider: aws
+    subscriptionSelector: "abcd-0001"
+status:
+    gardenerSecretName: subscription-secret
+```
+
+The resource structure is really simple, the Subscription Manager only needs to know the provider and a proper selector for the subscription to be selected. It provides credentials in a form of the Gardener secret name, which can be simply used as a parameter in the Shoot CR.
+
+## Cluster Provisioning Flow
+
+The flow from the user's perspective stays intact, the part that changes is the list of actions needed by KEB to make sure the cluster is up and running.
+
+First of all, after receiving the provisioning request, KEB should create a set of CRs for accordingly to the licence type of the Runtime:
+- For BYOC - ExternalCluster CR
+- For managed - GardenerCluster CR
+
+KEB should deliver all the required parameters for the cluster provisioning, this translates to the minimal infrastructure config for Gardener clusters, and the role/kubeconfig details for both types.
+
+Another thing that KEB would need to create is the Kyma CR for the Runtime, this will either require passing the proper secret reference so that kubeconfig can be accessed by the Lifecycle Manager.
+
+After the CRs are created, the respective Cluster Manager will pick up the provisioning request and will create a new Shoot for the runtime (managed scenario only), and provide a requested kubeconfig.
+
+At this point Lifecycle Manager can step in, it has all required resources to set up the Runtime:
+- running cluster
+- kubeconfig (passed via secret reference)
+- Kyma CR
+
+## Reacting to the kubeconfig rotation
+
+Since Lifecycle Manager and Module Manager need to interact with the provisioned K8S cluster directly, a mechanism for providing them with an up-to-date kubeconfig is needed.
 
-Worker pool implementation has to be revisited. Mothership should not manage dependencies - Component Reconciler should be written in a way that can recognize missing dependencies and fail fast. The next reconciliation invocation should happen when the configuration (Cluster Inventory) is updated (dependency). 
+There are few possible options here:
+- Watching K8S Secrets - Managers should watch the Secrets containing the kubeconfigs, with a big amount of Runtimes this may be resource intensive on the API server side.
```

Comment:
```suggestion
- Watching Kubernetes Secrets - Managers should watch the Secrets containing the kubeconfigs, with a big amount of Runtimes this may be resource intensive on the API server side.
```